Support other STM32WB target

> Same thing seems to be happening to me. 
> 
> Trying to use the new version of STM32duinoBLE 1.3.7 with a Generic STM32WB55VGYx board. 
> 
> Seems like the HCISharedMemTransportClass is never available?
> 
> 
> ```
> ~/Documents/Arduino/libraries/STM32duinoBLE/src/utility/HCISharedMemTransport.cpp:781:39: error: 'HCISharedMemTransport' was not declared in this scope; did you mean 'HCISharedMemTransportClass'?
>   781 | HCITransportInterface& HCITransport = HCISharedMemTransport;
>       |                                       ^~~~~~~~~~~~~~~~~~~~~
>       |                                       HCISharedMemTransportClass
> ```
> 
> it works if I revert back to an older version (ie. 1.2.9) 

 _Originally posted by @federica-ven in [#71](https://github.com/stm32duino/STM32duinoBLE/issues/71#issuecomment-2781545735)_

--- a/src/utility/HCISharedMemTransport.cpp
+++ b/src/utility/HCISharedMemTransport.cpp
@@ -22,11 +22,10 @@
 #include "STM32_WPAN/hw.h"
 #include "otp.h"
 
-#if defined(ARDUINO_NUCLEO_WB15CC) || defined(ARDUINO_P_NUCLEO_WB55RG) ||\
-      defined(ARDUINO_STM32WB5MM_DK) || defined(ARDUINO_P_NUCLEO_WB55_USB_DONGLE)
 HCISharedMemTransportClass HCISharedMemTransport;
-#else
-#error "Unsupported board or shield selected!"
+#if !defined(ARDUINO_NUCLEO_WB15CC) && !defined(ARDUINO_P_NUCLEO_WB55RG) &&\
+    !defined(ARDUINO_STM32WB5MM_DK) && !defined(ARDUINO_P_NUCLEO_WB55_USB_DONGLE)
+#warning "Selected board has never been tested with this library, ensure to have a correct configuration!"
 #endif
 
 /* Private variables ---------------------------------------------------------*/

